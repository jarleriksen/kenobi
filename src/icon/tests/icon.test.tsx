import React from 'react';
import { render } from '@testing-library/react';
import { Icon } from '../';
import '@testing-library/jest-dom/extend-expect';

describe('Icon', () => {
  test('it renders correctly', () => {
    const { getByTestId, asFragment } = render(
      <Icon icon="edit" title="test" />,
    );

    const icon = getByTestId('kenobi-icon');

    expect(icon).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  test('contains accessibility title', () => {
    const { getByTestId, asFragment } = render(
      <Icon icon="edit" title="test" />,
    );

    const icon = getByTestId('kenobi-icon');

    expect(icon).toContainHTML('<title>test</title>');
    expect(asFragment()).toMatchSnapshot();
  });
});
