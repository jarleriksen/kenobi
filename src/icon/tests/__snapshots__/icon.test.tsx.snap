// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon contains accessibility title 1`] = `
<DocumentFragment>
  <svg
    data-icon="edit"
    data-testid="kenobi-icon"
    height="16"
    style="fill: #425A70;"
    viewBox="0 0 16 16"
    width="16"
  >
    <title>
      test
    </title>
    <path
      d="M3.25 10.26l2.47 2.47 6.69-6.69-2.46-2.48-6.7 6.7zM.99 14.99l3.86-1.39-2.46-2.44-1.4 3.83zm12.25-14c-.48 0-.92.2-1.24.51l-1.44 1.44 2.47 2.47 1.44-1.44c.32-.32.51-.75.51-1.24.01-.95-.77-1.74-1.74-1.74z"
      fill-rule="evenodd"
    />
  </svg>
</DocumentFragment>
`;

exports[`Icon it renders correctly 1`] = `
<DocumentFragment>
  <svg
    data-icon="edit"
    data-testid="kenobi-icon"
    height="16"
    style="fill: #425A70;"
    viewBox="0 0 16 16"
    width="16"
  >
    <title>
      test
    </title>
    <path
      d="M3.25 10.26l2.47 2.47 6.69-6.69-2.46-2.48-6.7 6.7zM.99 14.99l3.86-1.39-2.46-2.44-1.4 3.83zm12.25-14c-.48 0-.92.2-1.24.51l-1.44 1.44 2.47 2.47 1.44-1.44c.32-.32.51-.75.51-1.24.01-.95-.77-1.74-1.74-1.74z"
      fill-rule="evenodd"
    />
  </svg>
</DocumentFragment>
`;
